posthog/tasks/alerts/utils.py

```diff
@@ -114,31 +117,91 @@ def next_check_time(alert: AlertConfiguration) -> datetime:
             raise ValueError(f"Invalid alert calculation interval: {alert.calculation_interval}")
 
 
-def send_notifications_for_breaches(alert: AlertConfiguration, breaches: list[str]) -> None:
-    subject = f"PostHog alert {alert.name} is firing"
-    campaign_key = f"alert-firing-notification-{alert.id}-{timezone.now().timestamp()}"
-    insight_url = f"/project/{alert.team.pk}/insights/{alert.insight.short_id}"
-    alert_url = f"{insight_url}?alert_id={alert.id}"
-    message = EmailMessage(
-        campaign_key=campaign_key,
-        subject=subject,
-        template_name="alert_check_firing",
-        template_context={
-            "match_descriptions": breaches,
-            "insight_url": insight_url,
-            "insight_name": alert.insight.name,
-            "alert_url": alert_url,
-            "alert_name": alert.name,
+def trigger_alert_hog_functions(alert: AlertConfiguration, event_name: str, properties: dict) -> None:
+    """Trigger all HogFunctions linked to the alert as notification destinations."""
+    invocation_globals = {
+        "alert": {
+            "id": str(alert.id),
+            "name": alert.name,
+            "insight_id": str(alert.insight_id),
+            "insight_short_id": alert.insight.short_id,
+            "state": alert.state,
+            "last_checked_at": alert.last_checked_at.isoformat() if alert.last_checked_at else None,
+        },
+        "event": {
+            "event": event_name,
+            "properties": {
+                "$alert_id": str(alert.id),
+                "$alert_name": alert.name,
+                **properties,  # Include specific properties like breaches or error
+            },
+            "timestamp": datetime.now(UTC).isoformat(),
         },
-    )
-    targets = alert.subscribed_users.all().values_list("email", flat=True)
-    if not targets:
-        raise RuntimeError(f"no targets configured for the alert {alert.id}")
-    for target in targets:
-        message.add_recipient(email=target)
+        "team_id": alert.team_id,
+    }
+
+    for hog_function in alert.notification_destinations.filter(enabled=True):
+        logger.info(
+            "Triggering HogFunction for alert notification",
+            alert_id=alert.id,
+            hog_function_id=hog_function.id,
+            hog_function_name=hog_function.name,
+            event_name=event_name,
+        )
+        try:
+            api_payload = {
+                "configuration": HogFunctionSerializer(hog_function).data,
+                "globals": invocation_globals,
+                "mock_async_functions": False,  # Ensure this is false for production
+            }
+
+            response = create_hog_invocation_test(
```

Comment:
@benjackwhite would you rather I use this or `produce_internal_event`, both eventually get picked up through cyclotron (that's what I gathered)

Reply:
this should never be used for anything critical. It is only for testing (hints in the name ;) )